src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -66,9 +76,19 @@ public void Dispose()
             }
         }
 
-        public Task StopAsync(CancellationToken cancellationToken)
+        public async Task StopAsync(CancellationToken cancellationToken)
         {
-            return host.UnregisterObserversAsync();
+            try
+            {
+                if (this.host != null && this.listernerStarted)
+                {
+                    await this.host.UnregisterObserversAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                this.trace.Error("Stopping the observer failed, potentially it was never started.", ex);
```

Comment:
Since this doesn't really bring anything down -- maybe we call this a Warning?

Reply:
Done